src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputOperationConverter.cs

```diff
@@ -85,17 +87,38 @@ public static InputOperation CreateOperation(ref Utf8JsonReader reader, string?
                 Console.Error.WriteLine($"[Warn]: InputOperation '{name}' does not have a description");
                 description = $"{name.Humanize()}.";
             }
+            crossLanguageDefinitionId = crossLanguageDefinitionId ?? throw new JsonException("InputOperation must have crossLanguageDefinitionId");
             uri = uri ?? throw new JsonException("InputOperation must have uri");
             path = path ?? throw new JsonException("InputOperation must have path");
             parameters = parameters ?? throw new JsonException("InputOperation must have parameters");
             responses = responses ?? throw new JsonException("InputOperation must have responses");
 
-            var inputOperation = new InputOperation(name, resourceName, summary, deprecated, description, accessibility, parameters, responses, httpMethod, requestBodyMediaType, uri, path, externalDocsUri, requestMediaTypes, bufferResponse, longRunning, paging, generateProtocolMethod, generateConvenienceMethod, keepClientDefaultValue);
+            var inputOperation = new InputOperation(name, resourceName, summary, deprecated, description, accessibility, parameters, responses, httpMethod, requestBodyMediaType, uri, path, externalDocsUri, requestMediaTypes, bufferResponse, longRunning, paging, generateProtocolMethod, generateConvenienceMethod, crossLanguageDefinitionId, keepClientDefaultValue)
+            {
+                IsNameChanged = IsNameChanged(crossLanguageDefinitionId, name)
+            };
             if (id != null)
             {
                 resolver.AddReference(id, inputOperation);
             }
             return inputOperation;
+
+            static bool IsNameChanged(string crossLanguageDefinitionId, string name)
+            {
+                var span = crossLanguageDefinitionId.AsSpan();
```

Comment:
Maybe we can get the IndexOf (.) first, then create the span and then check equal
```
int lastDotIndex = crossLanguageDefinitionId.LastIndexOf('.');
if (lastDotIndex > 0) {
  var idSpan = crossLanguageDefinitionId.AsSpan(lastDotIndex + 1);
  var nameSpan = name.AsSpan();
 if (nameSpan.Equals(idSpan, StringComparison.Ordinal))
  {
           return false;
   } else {
          return true;
   }
   
   return false;
}
```

Reply:
I used to get comments in analyzers form azure core team. They ask for `Span` v.s. `String` for performance consideration. I guess here we use `Span` for similar reason.

Reply:
yes. we should use span to get better performance. I write the code above may save a copy of string.